src/site/content/en/blog/pwa-url-handler/index.md

```diff
@@ -0,0 +1,291 @@
+
+```json
+{
+  "web_apps": [
+    {
+      "manifest": "https://music.example.com/manifest.webappmanifest",
+      "details": {
+        "paths": ["/*"],
+        "exclude_paths": ["/internal/*"]
+      }
+    }
+  ]
+}
+```
+
+{% Aside %} The `web-app-origin-association` file is similar to
+[Apple's associated domain file](https://developer.apple.com/documentation/safariservices/supporting_associated_domains_in_your_app#3001215),
+[Android's `assetlinks.json` file](https://developer.android.com/training/app-links/verify-site-associations),
+and
+[Windows' `windows-app-web-link` file](https://docs.microsoft.com/en-us/windows/uwp/launch-resume/web-to-app-linking#associate-your-app-and-website-with-a-json-file).
+What differs is that the `web-app-origin-association` file does not reference PWAs using a
+platform-specific app ID, but by their Web App Manifest URL. {% endAside %}
+
+#### When does a URL match?
+
+A PWA matches a URL for URL handling if both of the following conditions are fulfilled:
+
+- The URL matches one of the origin strings in `"url_handlers"`.
+- The browser is able to validate via the respective `web-app-origin-association` file that each
+  origin agrees to let this app handle such a URL.
+
+#### Regarding `web-app-origin-association` file discovery
+
+For the browser to discover the `web-app-origin-association` file, developers have two choices.
+
+- Add a `<link rel="web-app-origin-association">` element in the header section of the main document
+  at the origin's root path that points at the `web-app-origin-association` file via the `href`
+  attribute.
+- Alternatively, the `web-app-origin-association` file can also be placed in the
+  [`/.well-known/`](https://datatracker.ietf.org/doc/html/rfc8615) folder at the root level of the
+  app. For this to work, the file name must exactly be `web-app-origin-association`.
+
+## Demo
+
+To test PWAs as URL Handlers, be sure to [set the browser flag](#enabling-via-about:flags) as
+outlined above and then install the PWA at
+[https://mandymsft.github.io/pwa/](https://mandymsft.github.io/pwa/). By looking at its
+[Web App Manifest](https://github.com/mandymsft/pwa/blob/main/manifest.json), you can see that it
+handles URLs from the following URL patterns: `https://mandymsft.github.io` and
+`https://luhuangmsft.github.io`. Since the latter is on a different origin (`luhuangmsft.github.io`)
+than the PWA, the PWA on `mandymsft.github.io` needs to prove ownership, which happens via the
+`web-app-origin-association` file hosted at
+[https://luhuangmsft.github.io/.well-known/web-app-origin-association](https://luhuangmsft.github.io/.well-known/web-app-origin-association).
+
+To test that it is indeed working, send yourself a test message using an instant messaging app of
+your choice or an email that you view in an email client that is not web-based like Mail on macOS on
+your device with either of the links `https://mandymsft.github.io` or
```

Comment:
```suggestion
your choice or an email viewed in an email client that is not web-based like Mail on macOS. The email or text message should contain either of the links `https://mandymsft.github.io` or
```

Reply:
I had trouble following the sentence as it was. I took a shot at what I think you're saying.